sdk/core/Azure.Core/src/Pipeline/ActivityExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// HACK HACK HACK. Some runtime environments like Azure.Functions downgrade System.Diagnostic.DiagnosticSource package version causing method not found exceptions in customer apps
+    /// This type is a temporary workaround to avoid the issue
+    /// </summary>
+    internal static class ActivityExtensions
+    {
+        private static readonly MethodInfo? SetIdFormatMethod = typeof(Activity).GetMethod("SetIdFormat");
+        private static readonly MethodInfo? GetIdFormatMethod = typeof(Activity).GetProperty("IdFormat")?.GetMethod;
```

Comment:
What parameters is that method taking? If they add an overload to that method you will get an exception about an ambiguous method.

Reply:
It's a property, so no parameters.

Reply:
Ah yes!